added hardmod guide into the index (#1)
DS-Homebrew/wiki: pages/_en-US/ds-index/hardmod.md

  - A valid NAND backup of the DSi you are hardmodding
  - To be comfortable soldering to pads that are as small as 0.5mm in diameter

#### DSi side A pinouts
![DSi side A pinouts](/assets/images/dsi-hardmod/side_a.jpg)
#### DSi side B pinouts
![DSi side B pinouts](/assets/images/dsi-hardmod/side_b.png)
#### DSi XL side B pinouts
![DSi XL side B pinouts](/assets/images/dsi-hardmod/dsi_xl_side_b.png)

- You will need to solder the points on the motherboard to the microSD card adapter
  - CMD to pin 2
  - GND to pins 3 and 6
  - CLK to pin 5
  - DAT0 to pin 7

#### Soldered microSD adapter example
![microSD example](/assets/images/dsi-hardmod/sd.jpg)

- Insert the microSD card adapter into your computer
  - **WARNING** - Windows will ask you to format it: **DO NOT** format it - this might cause irrepairable damage

- You will first need to remove the NOCASH footer from the backup you are flashing to the DSi
- This can be done by running the hiyaCFW helper application found [here](https://github.com/mondul/HiyaCFW-Helper/releases/latest)

> 1. Download the version for the operating system you are using
> 1. Run the script, selecting the NAND backup you wish to flash to your DSi
> 1. Select the output path to be anywhere on your computer
> 
> - Once the script completes, there should be a folder at the selected output path with a long name starting with `FW`
>   - Use the file inside of this folder as the NAND image you flash onto the DSi

Reviewer: I just tried it myself and a couple things:
- Probably good to mention to switch to NAND mode, as that's not super obvious that you need to click that button
- For me it just outputs `08A1807916109128-no-footer.bin` in the path where the terminal is, so the home folder if you run it on macOS, does it ask for a path and make the FW... folder for you?

```suggestion
1. Download the version for the operating system you are using
1. Run the script, selecting the NAND backup you wish to flash to your DSi
1. Switch to NAND mode with the button to the left of the file path
1. Click `Start` to create the NAND backup without the no$gba footer

- Once the script completes, there should be a file with a long name ending in `-no-footer.bin` in the folder where your terminal is open
  - Use this file as the NAND image you flash onto the DSi
```

  - SD to microSD adapter
  - An SD card reader able to read eMMC chips running in single data-line mode. [Here's a known working one](https://www.amazon.com/dp/B006T9B6R2)
  - [Win32DiskImager](https://sourceforge.net/projects/win32diskimager/) and [HxD](https://mh-nexus.de/en/downloads.php?product=HxD20) if on Windows
  - A valid NAND backup of the DSi you are hardmodding
  - To be comfortable soldering to pads that are as small as 0.5mm in diameter

#### DSi side A pinouts
![DSi side A pinouts](/assets/images/dsi-hardmod/side_a.jpg)
#### DSi side B pinouts
![DSi side B pinouts](/assets/images/dsi-hardmod/side_b.png)
#### DSi XL side B pinouts
![DSi XL side B pinouts](/assets/images/dsi-hardmod/dsi_xl_side_b.png)

- You will need to solder the points on the motherboard to the microSD card adapter
  - CMD to pin 2
  - GND to pins 3 and 6
  - CLK to pin 5
  - DAT0 to pin 7

#### Soldered microSD adapter example
![microSD example](/assets/images/dsi-hardmod/sd.jpg)

- Insert the microSD card adapter into your computer
  - **WARNING** - Windows will ask you to format it: **DO NOT** format it - this might cause irrepairable damage

> - You will first need to remove the NOCASH footer from the backup you are flashing to the DSi
> - This can be done by running the hiyaCFW helper application found [here](https://github.com/mondul/HiyaCFW-Helper/releases/latest)

Reviewer: ![スクリーンショット 2021-01-12 6 50 30](https://user-images.githubusercontent.com/41608708/104316755-8f448680-54a2-11eb-9105-2dfba05a800b.png)

Just checked how it looks in the site and this blends in a bit so one more thing, I think it'd be better to give this a heading like this:
```suggestion
#### Removing the no$gba footer
You will first need to remove the NOCASH footer from the backup you are flashing to the DSi. This can be done by running the hiyaCFW helper application found [here](https://github.com/mondul/HiyaCFW-Helper/releases/latest)
```

Other than that this looks good to me so feel free to merge it, NightScript, if you think it's good

  - GND to pins 3 and 6
  - CLK to pin 5
  - DAT0 to pin 7

#### Soldered microSD adapter example
![microSD example](/assets/images/dsi-harmod/sd.jpg)

- Insert the microSD card adapter to your computer
  - **WARNING** - Windows will ask you to format it: **DO NOT** format it - this might cause irrepairable damage

#### Windows instructions (skip for macOS / Linux)

1. Open Win32DiskImager
2. Click the folder icon and browse to your desktop. In the text box, write NAND_0.bin. When selecting type, choose All types *.* 
3. Choose the device that is the DSi and click read. 
4. Once it is done, click the folder icon,change the name to NAND_1.bin and read again
5. Open HxD and drag both files into the editor. Go to the top bar, click "Analysis",click "File compare" from the drop down menu, then click "Compare".
6. Choose to compare both files then click OK when done
  - If it says "The chosen files are identical.", you are good to go to the next section
  - If it doesn't say that, and both NANDs aren't around 240MB, dump NAND_1/NAND_0 again
7. Open Win32DiskImager, click the folder icon and choose your existing NAND backup you created before.
8. Flash it with the `Write` button.

> 9. Unplug the SD card adapter and attempt to turn on the DSi

Reviewer: Minor thing, but I usually do markdown lists like
```md
1. thing
1. another thing
1. yet more things
```
as it auto does the numbers
1. thing
1. another thing
1. yet more things

#### Linux / macOS instructions (skip if on Windows)

1. Work out where the SD card is mounted
  - Linux:
    1. Unplug the SD card adapter
    2. Run `lbslk` on a terminal
    3. Plug in the SD card adapater
    4. Run `lbslk` again
    5. This time a new device should appear
    6. It may be called `/dev/sdb`, make sure you note down what it is called on *your* terminal

  - macOS:
    1. Unplug the SD card adapter
    2. Run `df` on a terminal
    3. Plug in the SD card adapter
    4. Run `df` again
    5. A new device should appear in the list

>     6. It may be called `/dev/disk1s1`, make sure you note down what it is called on *your* terminal/

Reviewer 2: Instead of makking this a new number step, try making this a sub-bullet of "A new device should appear in the list"

2. Dump the NAND
  - Run the following command on a terminal:
    - `cat {device-name} > nand0.bin` 
    - Replace {device-name} with the SD card mount location
    - E.g `cat /dev/sdb > nand0.bin`
    - Then run `cat {device-name} > nand1.bin`

3. Compare NAND dumps
  1. Run the following command:
    - Linux : `md5sum nand0.bin nand1.bin`
    - macOS : `md5 -r nand0.bin nand1.bin`
  2. Check that the hash generated matches each other.
  3. If not, redump the NAND, adjusting the wiring if necessary
  4. Keep on redumping the NAND until the hashes are identical

4. Flash NAND
  1. Locate the NAND backup you created before on the console.
  2. Run the following command:
    - `cat {existing-nand-backup} > {device-name}`
    - Replace `{existing-nand-backup}` with the location and name of your old NAND backup
    - Replace `{device-name}` with the mount point of the SD card
  3. At this point you may unplug the SD card adapter and attempt to turn on the DSi.

>   4. If all went well, the DSi should have booted to the state it was when the backup was created!

Reviewer 2: This isn't really a step. You should put this and the former one in a new line

Reviewer: fyi to highlight two lines you can click and drag the blue [+] thing

Reviewer 2: Interesting to note. Thank you

![Original DS motherboard with touch-points labeled](https://media.discordapp.net/attachments/547986366357700620/736370094392999947/mobo_pinout.png)

### Nintendo DSi

> - To hardmod a Nintendo DSi you will need:

Reviewer 2: This should not be a bullet point. Make it a standalone

>   - A DSi or DSi XL (LL)

Reviewer 2: Orly?

  - Pencil tipped soldering iron
  - Very fine wire, 28AWG or smaller, preferably 30AWG+
  - SD to microSD adapter

>   - An SD card reader able to read eMMC chips running in single data-line mode.

Reviewer 2: Might be worthwhile to add an Amazon link

  - [Win32DiskImager](https://sourceforge.net/projects/win32diskimager/) and [HxD](https://mh-nexus.de/en/downloads.php?product=HxD20) if on Windows
  - A valid DSi backup of the DSi you are hardmodding

>   - You must be comfortable soldering to pads that are as small as 0.5mm in diameter

Reviewer 2: "To hardmod a Nintendo DSi you will need:
   - You must be comfortable..."

>   - [Pencil tipped soldering iron](https://www.amazon.com/Sywon-Soldering-Station-Temperature-Adjustable/dp/B01N4571Q6/)
>   - [Very fine wire](https://www.amazon.com/BNTECHGO-Silicone-Flexible-Stranded-Impedance/dp/B01MXGNTA4/ref=sr_1_8?crid=1Y1M31ZLILOYN&dchild=1&keywords=30+awg+wire&qid=1608818117&sprefix=30+AWG+%2Caps%2C250&sr=8-8), 28AWG or smaller, preferably 30AWG+ 
>   - SD to microSD adapter
>   - An SD card reader able to read eMMC chips running in single data-line mode. [Here's a known working one](https://www.amazon.com/gp/product/B006T9B6R2)

Reviewer: ```suggestion
  - [Pencil tipped soldering iron](https://www.amazon.com/dp/B01N4571Q6)
  - [Very fine wire](https://www.amazon.com/dp/B01MXGNTA4), 28AWG or smaller, preferably 30AWG+ 
  - SD to microSD adapter
  - An SD card reader able to read eMMC chips running in single data-line mode. [Here's a known working one](https://www.amazon.com/dp/B006T9B6R2)
```
You don't need all the extra bits with the name and such for amazon links fyi

>   - A valid DSi backup of the DSi you are hardmodding

Reviewer: ```suggestion
  - A valid NAND backup of the DSi you are hardmodding
```

#### DSi side A pinouts
![DSi side A pinouts](https://safan41.github.io/hardmodguide/images/pinouts/DSi/a.jpg)
#### DSi side B pinouts
![DSi side B pinouts](https://safan41.github.io/hardmodguide/images/pinouts/DSi/b.png)
#### DSi XL side B pinouts

> ![DSi XL side B pinouts](https://safan41.github.io/hardmodguide/images/pinouts/DSi%20XL/nand.png)

Reviewer: Probably better to reupload these here under `/assets/images/` instead of just linking some other site that could change or be deleted or so btw

> ![DSi side B pinouts](/assets/images/dsi-hardmod/side_b.jpg)

Reviewer: ```suggestion
![DSi side B pinouts](/assets/images/dsi-hardmod/side_b.png)
```

- You will need to solder the points on the motherboard to the microSD card adapter
  - CMD to pin 2
  - GND to pins 3 and 6
  - CLK to pin 5
  - DAT0 to pin 7

#### Soldered microSD adapter example

> ![microSD example](/assets/images/dsi-harmod/sd.jpg)

Reviewer: ```suggestion
![microSD example](/assets/images/dsi-hardmod/sd.jpg)
```

> - Insert the microSD card adapter to your computer

Reviewer: ```suggestion
- Insert the microSD card adapter into your computer
```

  - **WARNING** - Windows will ask you to format it: **DO NOT** format it - this might cause irrepairable damage

#### Windows instructions (skip for macOS / Linux)

1. Open Win32DiskImager
1. Click the folder icon and browse to your desktop. In the text box, write NAND_0.bin. When selecting type, choose All types *.* 

> 1. Choose the device that is the DSi and click read. 

Reviewer: ```suggestion
1. Choose the device that is the DSi and click read
```

> 1. Once it is done, click the folder icon,change the name to NAND_1.bin and read again
> 1. Open HxD and drag both files into the editor. Go to the top bar, click "Analysis",click "File compare" from the drop down menu, then click "Compare".

Reviewer: ```suggestion
1. Once it is done, click the folder icon, change the name to NAND_1.bin and read again
1. Open HxD and drag both files into the editor. Go to the top bar, click "Analysis", click "File compare" from the drop down menu, then click "Compare".
```

1. Choose to compare both files then click OK when done
  - If it says "The chosen files are identical.", you are good to go to the next section
  - If it doesn't say that, and both NANDs aren't around 240MB, dump NAND_1/NAND_0 again

> 1. Open Win32DiskImager, click the folder icon and choose your existing NAND backup you created before.
> 1. Flash it with the `Write` button.

Reviewer: ```suggestion
1. Open Win32DiskImager, click the folder icon and choose your existing NAND backup you created before
1. Flash it with the `Write` button
```

1. Unplug the SD card adapter and attempt to turn on the DSi

#### Linux / macOS instructions (skip if on Windows)

1. Work out where the SD card is mounted
  - Linux:
    1. Unplug the SD card adapter
    1. Run `lbslk` on a terminal

>     1. Plug in the SD card adapater

Reviewer: ```suggestion
    1. Plug in the SD card adapter
```

    1. Run `lbslk` again
    1. This time a new device should appear

>       - It may be called `/dev/sdb`, make sure you note down what it is called on *your* terminal.

Reviewer: ```suggestion
      - It may be called `/dev/sdb`, make sure you note down what it is called on *your* terminal
```

  - macOS:
    1. Unplug the SD card adapter
    1. Run `df` on a terminal
    1. Plug in the SD card adapter
    1. Run `df` again
    1. A new device should appear in the list

>       - It may be called `/dev/disk1s1`, make sure you note down what it is called on *your* terminal.

Reviewer: ```suggestion
      - It may be called `/dev/disk1s1`, make sure you note down what it is called on *your* terminal
```

1. Dump the NAND
  - Run the following command on a terminal:
    - `cat {device-name} > nand0.bin` 
    - Replace {device-name} with the SD card mount location
    - E.g `cat /dev/sdb > nand0.bin`
    - Then run `cat {device-name} > nand1.bin`

1. Compare NAND dumps
  1. Run the following command:
    - Linux : `md5sum nand0.bin nand1.bin`
    - macOS : `md5 -r nand0.bin nand1.bin`

>   1. Check that the hash generated matches each other.

Reviewer: ```suggestion
  1. Check that the hash generated matches each other
```

  1. If not, redump the NAND, adjusting the wiring if necessary
  1. Keep on redumping the NAND until the hashes are identical

1. Flash NAND

>   1. Locate the NAND backup you created before on the console.

Reviewer: ```suggestion
  1. Locate the NAND backup you created before on the console
```
